### PR TITLE
update vision components to support all hyperparameters

### PR DESCRIPTION
**Pull request by sharma-riti**
- Adding all hyperparameters which are supported currently
- Remove enums for model names as it should be string
- Making all hyperparameters optional except for data and model name
- Update the Vision CE version to the latest

Test Runs with updated components - 
[Image Classification Run](https://ml.azure.com/experiments/id/4c73ed0f-69b1-40ff-9caf-cd836c2de901/runs/2ed14a4f-df1f-4851-a132-c83c1285bb0d?wsid=/subscriptions/381b38e9-9840-4719-a5a0-61d9585e1e91/resourceGroups/automlimage_eastus2_rg/providers/Microsoft.MachineLearningServices/workspaces/risha-ws-centraluseuap&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)
[Object Detection Run](https://ml.azure.com/experiments/id/4c73ed0f-69b1-40ff-9caf-cd836c2de901/runs/cf684403-88ce-46fc-add8-8b427fbe15b3?wsid=/subscriptions/381b38e9-9840-4719-a5a0-61d9585e1e91/resourceGroups/automlimage_eastus2_rg/providers/Microsoft.MachineLearningServices/workspaces/risha-ws-centraluseuap&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)
[Instance Segmentation Run](https://ml.azure.com/experiments/id/4c73ed0f-69b1-40ff-9caf-cd836c2de901/runs/48730673-f0b2-4e37-8dbc-17c87978e182?wsid=/subscriptions/381b38e9-9840-4719-a5a0-61d9585e1e91/resourceGroups/automlimage_eastus2_rg/providers/Microsoft.MachineLearningServices/workspaces/risha-ws-centraluseuap&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

### CODE REVIEW

File: training/vision/components/image_classification/spec.yaml [hunk 1]
```diff
@@ -79,23 +176,40 @@ outputs:
 
 code: ../src
 
-environment: azureml:AzureML-AutoML-DNN-Vision-GPU:98
+environment: azureml:AzureML-AutoML-DNN-Vision-GPU:107
```
Comment (skasturi) [inline training/vision/components/image_classification/spec.yaml:179]: @sharma-riti Are we expected to change this after every CE release?

**Comment by skasturi** [comment]
Are these different from the ones present here? 

[https://github.com/Azure/azureml-assets/tree/main/training/automl/components](https://github.com/Azure/azureml-assets/tree/main/training/automl/components)

File: training/vision/components/image_classification/spec.yaml (hunk 1 above)
Comment (sharma-riti) [reply training/vision/components/image_classification/spec.yaml:179]: We are still looking into the release process... for now wanted to upgrade to the latest CE... 

I am not very familiar with https://github.com/Azure/azureml-assets/tree/main/training/automl/components components and how they work ... maybe I'll sync with you and @novaturient95 to learn more and see if these are duplicates ...

File: training/vision/components/instance_segmentation/spec.yaml [hunk 2]
```diff
@@ -15,51 +15,187 @@ inputs:
...
+    type: integer
+    optional: true
+    min: 1
+  evaluation_frequency:
+    description: Frequency to evaluate validation dataset to get metric scores. Must be a positive integer.
+    type: integer
+    optional: true
+    min: 1
+  gradient_accumulation_step:
+    description: Gradient accumulation means running a configured number of gradient_accumulation_step without updating the model weights while accumulating the gradients of those steps, and then using the accumulated gradients to compute the weight updates. Must be a positive integer.
```
Comment (vivek-dani) [inline training/vision/components/instance_segmentation/spec.yaml:74]: Should we align things with image_classification_subgraph.yaml in AzureMlCli repo ?

File: training/vision/components/instance_segmentation/spec.yaml [hunk 3]
```diff
@@ -70,21 +206,47 @@ outputs:
 
 code: ../src
  
-environment: azureml:AzureML-AutoML-DNN-Vision-GPU:98
+environment: azureml:AzureML-AutoML-DNN-Vision-GPU:107
```
Comment (vivek-dani) [inline training/vision/components/instance_segmentation/spec.yaml:209]: is there a way to always take latest env ? (like if we remove the version no). Would it help if we always take latest env ?

File: training/vision/components/image_classification/spec.yaml (hunk 1 above)
Comment (sharma-riti) [reply training/vision/components/image_classification/spec.yaml]: Updated it to use the latest always.

Comment (sharma-riti) [reply training/vision/components/image_classification/spec.yaml]: Also, synced with @novaturient95 , these ones don't run the dpv2 flow but just the runtime code, so both the components are different.

File: training/vision/components/instance_segmentation/spec.yaml (hunk 2 above)
Comment (sharma-riti) [reply training/vision/components/instance_segmentation/spec.yaml]: done.

File: training/vision/components/instance_segmentation/spec.yaml (hunk 3 above)
Comment (sharma-riti) [reply training/vision/components/instance_segmentation/spec.yaml]: updated the env to use the latest always... here is the run - https://ml.azure.com/experiments/id/f6a41276-fe9e-4be4-875f-908db3add077/runs/06dee5bc-e00d-4674-bfa8-eadb234512c1?wsid=/subscriptions/381b38e9-9840-4719-a5a0-61d9585e1e91/resourceGroups/automlimage_eastus2_rg/providers/Microsoft.MachineLearningServices/workspaces/risha-ws-eastus2&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

File: training/vision/components/image_classification/spec.yaml [hunk 4]
```diff
@@ -5,72 +5,168 @@ description: Trains an image classification model. [Learn More](https://aka.ms/b
...
+    description: Path to MLTable for training data.
   validation_data:
     type: mltable
     optional: true
-    description: Validation data
+    description: Path to MLTable for validation data.
+  task_type:
+    description: Whether a single image can have multiple labels.
+    type: string
+    enum: ['image-classification', 'image-classification-multilabel']
```
Comment (daholste) [inline training/vision/components/image_classification/spec.yaml:21]: nit: may want to have tasks without hyphens displayed to the user? no idea, just thinking out loud

**Review by vivek-dani — APPROVED** [review]

File: training/vision/components/image_classification/spec.yaml (hunk 4 above)
Comment (sharma-riti) [reply training/vision/components/image_classification/spec.yaml:21]: JOS/backend will send out these existing task types... also current users sdk/cli should be familiar with these types... so wanted to use them as is..

File: training/vision/components/image_classification/spec.yaml [hunk 5]
```diff
@@ -5,72 +5,168 @@ description: Trains an image classification model. [Learn More](https://aka.ms/b
...
     type: integer
     optional: true
     min: 1
+  validation_resize_size:
+    description: Image size to which to resize before cropping for validation dataset. Note - seresnext doesn't take an arbitrary size.
+    type: integer
+    optional: true
+    min: 1
+  warmup_cosine_lr_cycles:
+    description: Value of cosine cycle when learning rate scheduler is warmup_cosine. Please check for https://learn.microsoft.com/azure/machine-learning/reference-automl-images-hyperparameters more information.
```
Comment (daholste) [inline training/vision/components/image_classification/spec.yaml:150]: nit: wonder if URL needs to be or could be hyperlinked in description? not a big deal, just thinking out loud!

**Review by daholste — APPROVED** [review]

File: training/vision/components/image_classification/spec.yaml (hunk 5 above)
Comment (sharma-riti) [reply training/vision/components/image_classification/spec.yaml:150]: synced offline..

**Pull request merged**